From 01fd4064768d0bd430ae7529bf716219ea5f6418 Mon Sep 17 00:00:00 2001
From: sonicxml <sonicxml@gmail.com>
Date: Fri, 20 Jul 2012 17:15:12 -0400
Subject: [PATCH] Fix long press back errors

---
 .../internal/policy/impl/PhoneWindowManager.java        |   15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
index a1b6209..451ca7b 100755
--- a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
+++ b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
@@ -1921,6 +1921,14 @@ public long interceptKeyBeforeDispatching(WindowState win, KeyEvent event, int p
                 showOrHideRecentAppsDialog(RECENT_APPS_BEHAVIOR_SHOW_OR_DISMISS);
             }
             return -1;
+
+	} else if (keyCode == KeyEvent.KEYCODE_BACK) {
+            	if (Settings.Secure.getInt(mContext.getContentResolver(),
+			Settings.Secure.KILL_APP_LONGPRESS_BACK, 0) == 1) {
+                	if (down && repeatCount == 0) {
+                    		mHandler.postDelayed(mBackLongPress, 2000);
+                	}
+		}
         } else if (keyCode == KeyEvent.KEYCODE_ASSIST) {
             if (down) {
                 if (repeatCount == 0) {
@@ -1931,12 +1939,7 @@ public long interceptKeyBeforeDispatching(WindowState win, KeyEvent event, int p
                          launchAssistLongPressAction();
                     }
                 }
-            } else if (keyCode == KeyEvent.KEYCODE_BACK) {
-            	if (Settings.Secure.getInt(mContext.getContentResolver(),
-                    Settings.Secure.KILL_APP_LONGPRESS_BACK, 0) == 1) {
-                	if (down && repeatCount == 0) {
-                    		mHandler.postDelayed(mBackLongPress, 2000);
-                }
+
             } else {
                 if (mAssistKeyLongPressed) {
                     mAssistKeyLongPressed = false;
-- 
1.7.10

